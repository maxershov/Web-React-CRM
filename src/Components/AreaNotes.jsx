import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { ChangeProfileValue, addNewDayDataToJSON } from '../App';


const AreaNotes = (props) => {
  const { notesValue, type, dayObject } = props;
  const [notesData, setNotesData] = useState(notesValue);
  const { codeLink } = useParams();

  const saveNotes = (event) => {
    if (event.key === 'Enter' || event.target.id === 'clickNotes') {
      if (type === 'PERSON') {
        ChangeProfileValue(codeLink, notesData, "notes")
      } else if (type === 'DAY_DATA') {
        dayObject.notes = notesData;
        addNewDayDataToJSON(dayObject);
      }
    }
  }
  useEffect(() => {
    setNotesData(notesValue);
  }, [notesValue]);
  return (
    <>
      <textarea onChange={event => setNotesData(event.target.value)} onKeyDown={saveNotes} value={notesData} />
      <button type="button" id="clickNotes" onClick={saveNotes}>Изменить</button>
    </>
  )
}

export default AreaNotes;