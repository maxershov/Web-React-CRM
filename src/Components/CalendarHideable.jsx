import React, { useState } from 'react';
import { useParams } from 'react-router-dom';
import { format} from 'date-fns'
import Calendar from 'react-calendar/dist/entry.nostyle';
import { ChangeProfileValue } from '../App';


const CalendarHideable = (props) => {
  const [renderCalendar, setRenderCalendar] = useState('none');
  const { codeLink } = useParams();

  function changeDate(dateTo) {
    const date = format(dateTo, 'dd-MM-yyyy');
    props.dateType === 'setParent' ?
      props.setParentDate(date) :
      ChangeProfileValue(codeLink, date, props.dateType);
    setRenderCalendar('none');
  }

  function deleteDate() {
    ChangeProfileValue(codeLink, "", props.dateType);
    setRenderCalendar('none');
  }
  return (
    <>
      <div className={`${props.dateType}Field`}>
        <label>{props.сalendarName}</label>
        <input onClick={() => setRenderCalendar('block')} type="text" readOnly value={props.date} />
      </div>
      <div style={{ display: renderCalendar }} className="calendar" id="calendar">
        <Calendar className="calendar" onChange={date => changeDate(date)} />
        <button type="button" onClick={() => setRenderCalendar('none')}>Убрать календарь</button>
        <button type="button" onClick={() => deleteDate()}>Удалить дату</button>
      </div>
    </>
  );
}

export default CalendarHideable;