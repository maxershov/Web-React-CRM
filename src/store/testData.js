import { format } from 'date-fns'
export const dayData = [
  {
    date: format(new Date(),"dd-MM-yyyy"),
    notes: "Заметки меняются при изменении даты",
    history: []
  }
];

export const activityData = [
  {
    code: "1",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "2",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "3",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "4",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "5",
    activity: []
  },
  {
    code: "6",
    activity: []
  },
  {
    code: "7",
    activity: []
  },
  {
    code: "8",
    activity: []
  },
  {
    code: "9",
    activity: []
  },
  {
    code: "10",
    activity: []
  },
  {
    code: "11",
    activity: []
  },
  {
    code: "12",
    activity: []
  },
  {
    code: "13",
    activity: []
  },
  {
    code: "14",
    activity: []
  },
  {
    code: "15",
    activity: []
  },
  {
    code: "16",
    activity: []
  },
  {
    code: "17",
    activity: []
  },
  {
    code: "18",
    activity: []
  },
  {
    code: "19",
    activity: []
  },
  {
    code: "20",
    activity: []
  },
  {
    code: "21",
    activity: []
  },
  {
    code: "22",
    activity: []
  },
  {
    code: "23",
    activity: []
  },
  {
    code: "24",
    activity: []
  },
  {
    code: "25",
    activity: []
  },
  {
    code: "26",
    activity: []
  },
  {
    code: "27",
    activity: []
  },
  {
    code: "28",
    activity: []
  },
  {
    code: "29",
    activity: []
  },
  {
    code: "30",
    activity: []
  },
  {
    code: "31",
    activity: []
  },
  {
    code: "32",
    activity: []
  },
  {
    code: "33",
    activity: []
  },
  {
    code: "34",
    activity: []
  },
  {
    code: "35",
    activity: []
  },
  {
    code: "36",
    activity: []
  },
  {
    code: "37",
    activity: []
  },
  {
    code: "38",
    activity: []
  },
  {
    code: "39",
    activity: []
  },
  {
    code: "40",
    activity: []
  },
  {
    code: "41",
    activity: []
  },
  {
    code: "42",
    activity: []
  },
  {
    code: "43",
    activity: []
  },
  {
    code: "44",
    activity: []
  },
  {
    code: "45",
    activity: []
  },
  {
    code: "46",
    activity: []
  },
  {
    code: "47",
    activity: []
  },
  {
    code: "48",
    activity: []
  },
  {
    code: "49",
    activity: []
  },
  {
    code: "50",
    activity: []
  }
];

export const personData = [
  {
    personName: "Murphy Mccarthy",
    contract: "МИНИГРУППА",
    dateBirth: "17-10-1979",
    telNum: "1-256-471-0997",
    code: "1",
    autoMonth: 4,
    remain: "10",
    days: "",
    photoId: 3,
    rent: "21-04-2020",
    deposite: 3699,
    notes: "diam. Pellentesque habitant morbi tristique"
  },
  {
    personName: "Garrett Howe",
    contract: "6 ГРУПП",
    dateBirth: "09-06-1994",
    telNum: "1-843-886-7481",
    code: "2",
    autoMonth: 7,
    remain: "",
    days: "21-05-2020",
    photoId: 6,
    rent: "26-10-2019",
    deposite: 1085,
    notes: "magna. Phasellus dolor elit, pellentesque"
  },
  {
    personName: "Hilel Castaneda",
    contract: "3 ГРУПП",
    dateBirth: "11-03-1988",
    telNum: "1-737-288-1405",
    code: "3",
    autoMonth: 2,
    remain: "",
    days: "13-02-2020",
    photoId: 0,
    rent: "26-09-2019",
    deposite: 3669,
    notes: "ut quam vel sapien imperdiet"
  },
  {
    personName: "Rhoda Webb",
    contract: "1 СВОБ",
    dateBirth: "02-04-1967",
    telNum: "1-870-541-1034",
    code: "4",
    autoMonth: 8,
    remain: "",
    days: "27-06-2020",
    photoId: 0,
    rent: "15-10-2019",
    deposite: 303,
    notes: "auctor ullamcorper, nisl arcu iaculis"
  },
  {
    personName: "Kylie Blackburn",
    contract: "ЛИД",
    dateBirth: "04-07-1981",
    telNum: "1-562-298-7788",
    code: "5",
    autoMonth: 9,
    remain: 12,
    days: "24-10-2019",
    photoId: 8,
    rent: "12-01-2020",
    deposite: 3724,
    notes: "Curabitur sed tortor. Integer aliquam"
  },
  {
    personName: "Autumn Swanson",
    contract: "ЛИД",
    dateBirth: "03-10-1983",
    telNum: "1-880-353-9575",
    code: "6",
    autoMonth: 1,
    remain: 9,
    days: "17-11-2019",
    photoId: 0,
    rent: "02-11-2019",
    deposite: 4289,
    notes: "In scelerisque scelerisque dui. Suspendisse"
  },
  {
    personName: "Montana Kaufman",
    contract: "СОТРУДНИК",
    dateBirth: "21-11-1981",
    telNum: "1-973-533-7140",
    code: "7",
    autoMonth: 8,
    remain: 10,
    days: "25-10-2020",
    photoId: 7,
    rent: "08-10-2019",
    deposite: 1588,
    notes: "Praesent interdum ligula eu enim."
  },
  {
    personName: "Otto Ryan",
    contract: "НЕТ",
    dateBirth: "27-07-1963",
    telNum: "1-892-415-9444",
    code: "8",
    autoMonth: 3,
    remain: 8,
    days: "29-05-2019",
    photoId: 10,
    rent: "03-09-2019",
    deposite: 4755,
    notes: "aliquet, sem ut cursus luctus,"
  },
  {
    personName: "Karleigh Hurley",
    contract: "НЕТ",
    dateBirth: "23-01-1954",
    telNum: "1-844-835-1769",
    code: "9",
    autoMonth: 1,
    remain: 13,
    days: "08-03-2019",
    photoId: 0,
    rent: "25-09-2019",
    deposite: 1454,
    notes: "placerat, orci lacus vestibulum lorem,"
  },
  {
    personName: "Rebekah English",
    contract: "СОТРУДНИК",
    dateBirth: "15-05-1978",
    telNum: "1-893-501-2845",
    code: "10",
    autoMonth: 8,
    remain: 22,
    days: "08-01-2021",
    photoId: 9,
    rent: "01-12-2019",
    deposite: 2972,
    notes: "luctus et ultrices posuere cubilia"
  },
  {
    personName: "Francis Cantu",
    contract: "ЛИД",
    dateBirth: "29-02-1972",
    telNum: "1-218-584-2826",
    code: "11",
    autoMonth: 7,
    remain: 24,
    days: "21-12-2019",
    photoId: 0,
    rent: "21-02-2019",
    deposite: 2790,
    notes: "velit. Quisque varius. Nam porttitor"
  },
  {
    personName: "Raja Rowland",
    contract: "12 СВОБ",
    dateBirth: "05-11-1991",
    telNum: "1-630-349-1486",
    code: "12",
    autoMonth: 8,
    remain: "",
    days: "07-05-2019",
    photoId: 0,
    rent: "16-03-2019",
    deposite: 3750,
    notes: "ac facilisis facilisis, magna tellus"
  },
  {
    personName: "Mufutau Price",
    contract: "3 КФ",
    dateBirth: "26-10-1961",
    telNum: "1-571-985-3972",
    code: "13",
    autoMonth: 5,
    remain: "",
    days: "26-04-2019",
    photoId: 0,
    rent: "07-09-2020",
    deposite: -123,
    notes: "sed, facilisis vitae, orci. Phasellus"
  },
  {
    personName: "Tate Kerr",
    contract: "НЕТ",
    dateBirth: "25-07-1983",
    telNum: "1-736-691-3234",
    code: "14",
    autoMonth: 1,
    remain: 9,
    days: "25-01-2020",
    photoId: 0,
    rent: "14-10-2019",
    deposite: 1388,
    notes: "non, vestibulum nec, euismod in,"
  },
  {
    personName: "Nichole Tran",
    contract: "СОТРУДНИК",
    dateBirth: "09-11-1980",
    telNum: "1-977-432-7973",
    code: "15",
    autoMonth: 5,
    remain: 23,
    days: "11-10-2020",
    photoId: 11,
    rent: "22-01-2021",
    deposite: 4916,
    notes: "arcu. Vestibulum ut eros non"
  },
  {
    personName: "Kibo Hayden",
    contract: "НЕТ",
    dateBirth: "11-07-1987",
    telNum: "1-701-574-1974",
    code: "16",
    autoMonth: 7,
    remain: 7,
    days: "09-01-2020",
    photoId: 0,
    rent: "30-11-2019",
    deposite: 3092,
    notes: "odio a purus. Duis elementum,"
  },
  {
    personName: "McKenzie Phillips",
    contract: "1 СВОБ",
    dateBirth: "22-09-1956",
    telNum: "1-506-702-6936",
    code: "17",
    autoMonth: 8,
    remain: "",
    days: "05-08-2019",
    photoId: 4,
    rent: "06-04-2019",
    deposite: 351,
    notes: "elementum purus, accumsan interdum libero"
  },
  {
    personName: "TaShya Moore",
    contract: "1 СВОБ",
    dateBirth: "10-10-1965",
    telNum: "1-115-227-9451",
    code: "18",
    autoMonth: 8,
    remain: "",
    days: "04-11-2019",
    photoId: 0,
    rent: "07-11-2020",
    deposite: 1003,
    notes: "fringilla cursus purus. Nullam scelerisque"
  },
  {
    personName: "Holly Bradley",
    contract: "6 КФ",
    dateBirth: "",
    telNum: "",
    code: "19",
    autoMonth: 8,
    remain: "",
    days: "17-02-2019",
    photoId: 2,
    rent: "26-02-2019",
    deposite: 2104,
    notes: "lectus ante dictum mi, ac"
  },
  {
    personName: "Xander Moses",
    contract: "МИНИГРУППА",
    dateBirth: "",
    telNum: "1-942-292-3642",
    code: "20",
    autoMonth: 9,
    remain: 14,
    days: "30-11-2019",
    photoId: 5,
    rent: "07-01-2020",
    deposite: 3493,
    notes: "sapien molestie orci tincidunt adipiscing."
  },
  {
    personName: "Amaya Valentine",
    contract: "ПТ",
    dateBirth: "10-11-1953",
    telNum: "1-740-709-8222",
    code: "21",
    autoMonth: 5,
    remain: 18,
    days: "26-11-2019",
    photoId: 0,
    rent: "06-06-2020",
    deposite: 3790,
    notes: "porttitor eros nec tellus. Nunc"
  },
  {
    personName: "Thane Alvarado",
    contract: "ПТ",
    dateBirth: "01-10-1982",
    telNum: "1-872-104-7996",
    code: "22",
    autoMonth: 6,
    remain: 22,
    days: "03-02-2019",
    photoId: 0,
    rent: "25-03-2020",
    deposite: 1335,
    notes: "lobortis augue scelerisque mollis. Phasellus"
  },
  {
    personName: "Coby Mejia",
    contract: "ПТ",
    dateBirth: "13-11-1976",
    telNum: "1-372-186-7017",
    code: "23",
    autoMonth: 8,
    remain: 19,
    days: "06-03-2020",
    photoId: 0,
    rent: "05-02-2019",
    deposite: 2935,
    notes: "non enim commodo hendrerit. Donec"
  },
  {
    personName: "Travis Mccullough",
    contract: "ПТ",
    dateBirth: "15-09-1987",
    telNum: "1-743-426-2899",
    code: "24",
    autoMonth: 8,
    remain: 4,
    days: "19-08-2020",
    photoId: 0,
    rent: "04-05-2020",
    deposite: 3009,
    notes: "conubia nostra, per inceptos hymenaeos."
  },
  {
    personName: "George Kaufman",
    contract: "3 КФ",
    dateBirth: "03-02-1948",
    telNum: "1-140-514-2906",
    code: "25",
    autoMonth: 8,
    remain: "",
    days: "27-10-2020",
    photoId: 0,
    rent: "13-07-2019",
    deposite: 1141,
    notes: "nascetur ridiculus mus. Aenean eget"
  },
  {
    personName: "Adrienne Craft",
    contract: "ЛИД",
    dateBirth: "02-05-1981",
    telNum: "1-974-412-0982",
    code: "26",
    autoMonth: 5,
    remain: 3,
    days: "29-11-2019",
    photoId: 0,
    rent: "24-10-2019",
    deposite: 4554,
    notes: "tellus. Suspendisse sed dolor. Fusce"
  },
  {
    personName: "Cain Roberson",
    contract: "НЕТ",
    dateBirth: "10-04-1965",
    telNum: "1-747-108-3057",
    code: "27",
    autoMonth: 1,
    remain: 25,
    days: "14-12-2019",
    photoId: 0,
    rent: "14-05-2019",
    deposite: 1530,
    notes: "Vestibulum ante ipsum primis in"
  },
  {
    personName: "Ivana Carey",
    contract: "СОТРУДНИК",
    dateBirth: "27-12-1944",
    telNum: "1-822-483-5692",
    code: "28",
    autoMonth: 9,
    remain: 8,
    days: "15-04-2019",
    photoId: 0,
    rent: "23-02-2020",
    deposite: 2230,
    notes: "taciti sociosqu ad litora torquent"
  },
  {
    personName: "Merritt Becker",
    contract: "12 СВОБ",
    dateBirth: "07-10-1953",
    telNum: "1-414-164-3453",
    code: "29",
    autoMonth: 1,
    remain: "",
    days: "28-01-2019",
    photoId: 0,
    rent: "01-02-2019",
    deposite: -205,
    notes: "velit. Cras lorem lorem, luctus"
  },
  {
    personName: "Shay Randolph",
    contract: "ПТ",
    dateBirth: "28-03-2004",
    telNum: "1-498-448-2146",
    code: "30",
    autoMonth: 3,
    remain: 10,
    days: "21-02-2020",
    photoId: 0,
    rent: "26-04-2019",
    deposite: 82,
    notes: "rutrum lorem ac risus. Morbi"
  },
  {
    personName: "Isaiah Sweet",
    contract: "3 КФ",
    dateBirth: "27-01-1981",
    telNum: "1-404-934-8475",
    code: "31",
    autoMonth: 9,
    remain: "",
    days: "01-12-2019",
    photoId: 0,
    rent: "22-01-2020",
    deposite: 2543,
    notes: "cursus luctus, ipsum leo elementum"
  },
  {
    personName: "Caleb Washington",
    contract: "12 КФ",
    dateBirth: "02-03-1975",
    telNum: "1-437-259-5837",
    code: "32",
    autoMonth: 5,
    remain: 5,
    days: "31-10-2020",
    photoId: 0,
    rent: "03-02-2020",
    deposite: 3991,
    notes: "quis diam. Pellentesque habitant morbi"
  },
  {
    personName: "Christen Dominguez",
    contract: "МИНИГРУППА",
    dateBirth: "30-03-1974",
    telNum: "1-266-983-7881",
    code: "33",
    autoMonth: 5,
    remain: 19,
    days: "02-01-2021",
    photoId: 0,
    rent: "17-12-2019",
    deposite: 564,
    notes: "Fusce mollis. Duis sit amet"
  },
  {
    personName: "Reese Barry",
    contract: "3 ТА",
    dateBirth: "12-01-1964",
    telNum: "1-169-703-3860",
    code: "34",
    autoMonth: 9,
    remain: "",
    days: "05-10-2020",
    photoId: 0,
    rent: "17-08-2019",
    deposite: 2498,
    notes: "posuere at, velit. Cras lorem"
  },
  {
    personName: "Flynn Moran",
    contract: "3 ТА",
    dateBirth: "23-01-1982",
    telNum: "1-623-206-4393",
    code: "35",
    autoMonth: 7,
    remain: "",
    days: "29-02-2020",
    photoId: 0,
    rent: "05-09-2019",
    deposite: 2654,
    notes: "et ipsum cursus vestibulum. Mauris"
  },
  {
    personName: "Tashya Faulkner",
    contract: "1 СВОБ",
    dateBirth: "06-06-1948",
    telNum: "1-726-285-5889",
    code: "36",
    autoMonth: 6,
    remain: "",
    days: "27-09-2020",
    photoId: 0,
    rent: "26-07-2019",
    deposite: 2232,
    notes: "elit, a feugiat tellus lorem"
  },
  {
    personName: "April Lynn",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "23-08-1953",
    telNum: "1-646-749-0720",
    code: "37",
    autoMonth: 6,
    remain: 22,
    days: "05-03-2019",
    photoId: 0,
    rent: "31-12-2020",
    deposite: 2731,
    notes: "tincidunt pede ac urna. Ut"
  },
  {
    personName: "Pearl Nguyen",
    contract: "3 КФ",
    dateBirth: "25-12-1978",
    telNum: "1-233-110-5293",
    code: "38",
    autoMonth: 7,
    remain: "",
    days: "14-08-2020",
    photoId: 0,
    rent: "14-08-2019",
    deposite: 2941,
    notes: "eget metus eu erat semper"
  },
  {
    personName: "Lael Cervantes",
    contract: "ЛИД",
    dateBirth: "26-07-1995",
    telNum: "1-825-543-2207",
    code: "39",
    autoMonth: 4,
    remain: 7,
    days: "22-03-2019",
    photoId: 0,
    rent: "23-12-2019",
    deposite: 1166,
    notes: "ultrices posuere cubilia Curae; Donec"
  },
  {
    personName: "Asher Puckett",
    contract: "НЕТ",
    dateBirth: "18-09-1963",
    telNum: "1-690-233-3770",
    code: "40",
    autoMonth: 9,
    remain: 4,
    days: "24-01-2020",
    photoId: 0,
    rent: "07-12-2019",
    deposite: 4218,
    notes: "nisi. Aenean eget metus. In"
  },
  {
    personName: "Miriam Levine",
    contract: "СОТРУДНИК",
    dateBirth: "09-08-1959",
    telNum: "1-783-837-2369",
    code: "41",
    autoMonth: 8,
    remain: 9,
    days: "06-05-2019",
    photoId: 0,
    rent: "20-10-2020",
    deposite: 769,
    notes: "vestibulum. Mauris magna. Duis dignissim"
  },
  {
    personName: "Arsenio Hendricks",
    contract: "СОТРУДНИК",
    dateBirth: "16-08-1970",
    telNum: "1-152-271-3796",
    code: "42",
    autoMonth: 2,
    remain: 8,
    days: "31-07-2020",
    photoId: 0,
    rent: "11-03-2020",
    deposite: -196,
    notes: "adipiscing non, luctus sit amet,"
  },
  {
    personName: "Leilani Wheeler",
    contract: "СОТРУДНИК",
    dateBirth: "27-01-1973",
    telNum: "1-739-409-1958",
    code: "43",
    autoMonth: 1,
    remain: 13,
    days: "13-01-2021",
    photoId: 0,
    rent: "19-08-2020",
    deposite: 2852,
    notes: "interdum. Sed auctor odio a"
  },
  {
    personName: "Christen Sloan",
    contract: "6 ТА",
    dateBirth: "20-06-1968",
    telNum: "1-507-644-9918",
    code: "44",
    autoMonth: 4,
    remain: "",
    days: "12-11-2019",
    photoId: 0,
    rent: "29-12-2020",
    deposite: 3521,
    notes: "nunc, ullamcorper eu, euismod ac,"
  },
  {
    personName: "Lev Hess",
    contract: "СОТРУДНИК",
    dateBirth: "12-10-1955",
    telNum: "1-262-360-4575",
    code: "45",
    autoMonth: 7,
    remain: "",
    days: "10-06-2020",
    photoId: 0,
    rent: "10-06-2019",
    deposite: 4461,
    notes: "malesuada vel, venenatis vel, faucibus"
  },
  {
    personName: "Jasper Mills",
    contract: "3 КФ",
    dateBirth: "10-12-1994",
    telNum: "1-634-953-8909",
    code: "46",
    autoMonth: 3,
    remain: "",
    days: "29-12-2019",
    photoId: 0,
    rent: "07-10-2019",
    deposite: 755,
    notes: "consectetuer adipiscing elit. Curabitur sed"
  },
  {
    personName: "Hyatt Johnston",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "04-12-2004",
    telNum: "1-555-735-5064",
    code: "47",
    autoMonth: 8,
    remain: 10,
    days: "25-06-2019",
    photoId: 0,
    rent: "18-01-2021",
    deposite: 2133,
    notes: "arcu. Sed et libero. Proin"
  },
  {
    personName: "Gloria Dixon",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "02-07-1952",
    telNum: "1-553-504-4377",
    code: "48",
    autoMonth: 5,
    remain: 18,
    days: "26-11-2020",
    photoId: 0,
    rent: "09-09-2019",
    deposite: -77,
    notes: "eget varius ultrices, mauris ipsum"
  },
  {
    personName: "Jonah Dunlap",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "20-06-1953",
    telNum: "1-967-741-9273",
    code: "49",
    autoMonth: 5,
    remain: 18,
    days: "04-10-2020",
    photoId: 0,
    rent: "15-02-2020",
    deposite: 2028,
    notes: "diam. Proin dolor. Nulla semper"
  },
  {
    personName: "Wesley White",
    contract: "НЕТ",
    dateBirth: "23-04-1993",
    telNum: "1-522-349-9026",
    code: "50",
    autoMonth: 5,
    remain: 24,
    days: "30-12-2019",
    photoId: 0,
    rent: "23-03-2019",
    deposite: 3865,
    notes: "ut ipsum ac mi eleifend"
  }
];
