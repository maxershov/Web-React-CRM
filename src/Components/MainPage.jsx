import React, { useState } from 'react';
import { connect } from 'react-redux';
import { Link, useHistory } from "react-router-dom";
import { format, parse } from 'date-fns'
import ReactTable from 'react-table-6/react-table.min';
import Calendar from 'react-calendar/dist/entry.nostyle';
import AreaNotes from './AreaNotes';
import CodeScanner from './CodeScanner'
import FormData from './FormData';
import { getIndexByCode, getDateObj  } from '../App';


// set width to table colums by .className size
function widthForTable(value) {
  return Math.round(window.innerWidth * (value / 100))
}


function isToday(date){
  const todayDate  = format(new Date(),'dd-MM-yyyy')
  return todayDate === date;
}


export const MainPage = (props) => {
  const history = useHistory();
  const personData = JSON.parse(props.personData);
  const [loadedDate, setLoadedDate] = useState(format(new Date(),'dd-MM-yyyy'));
  const data = JSON.parse(getDateObj(loadedDate));
  
  const changeLoadDate = (date) => {
    const formatedDate = format(date, 'dd-MM-yyyy');
    setLoadedDate(formatedDate);
  }

  function getPhotoId(code) {
    return personData[getIndexByCode(code)].photoId
  }

  return (
    <>
      <div className="mainPage">
        <Calendar className="calendar calendarMain" value={parse(loadedDate, 'dd-MM-yyyy', new Date())} onChange={(date) => changeLoadDate(date)} />
        <div className="notesMain font-white-shadow"><AreaNotes notesValue={data.notes} type="DAY_DATA" dayObject={data} /></div>
        <div className="newProfileField"><FormData baseValue="" formLabel="Новый профиль:" type="NEW_PERSON" route={history} /></div>
        {isToday(loadedDate) 
        ? <div className="newCodeField"><CodeScanner dayObject={data} date={loadedDate} /></div> : undefined}
      </div>
      <div className="tableMain">
        <ReactTable
          className="table -striped -highlight"
          previousText="Назад"
          nextText="Вперед"
          loadingText="Загрузка"
          noDataText="Нет данных"
          pageText="Страница"
          ofText="из"
          rowsText="профилей"
          data={data.history}
          columns={[
            {
              Header: 'Фото',
              width: widthForTable(20),
              accessor: 'code',
              headerClassName: 'tableHeader',
              Cell: ({ value }) => (
                <button type="button" onClick={() => history.push(`/profile/${  value}`)}><img id="tablePhoto" alt="tablePhoto" height={80} src={require(`../images/${getPhotoId(value)}.jpg`)} /></button>)                
            },
            {
              Header: 'Имя',
              accessor: 'code',
              width: widthForTable(60),
              headerClassName: 'tableHeader',
              style: { whiteSpace: 'unset' },
              Cell: ({value}) => (<Link to={`/profile/${value}`}>{personData[getIndexByCode(value)].personName}</Link>)
            },
            {
              Header: 'Время',
              width: widthForTable(20),
              accessor: 'time',
              headerClassName: 'tableHeader',
            }]}
          defaultSorted={[{
            id: 'time',
            desc: true
          }]}
          defaultPageSize={5}
        />
      </div>
    </>
  );
}


const mapStateToProps = state => {
  return {
    personData: state.personStore.data,
    dayData: state.dayDataStore.data
  }
}

export default connect(mapStateToProps)(MainPage);