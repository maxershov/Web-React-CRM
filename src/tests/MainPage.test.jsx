import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MainPage } from '../Components/MainPage'
import { persons } from './testData/testData'

// Set mock for router func => get url for navigation
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

// Set default date to make all snapshots equal
const defaultDate = new Date(0);
jest.spyOn(global, 'Date').mockImplementation(() => defaultDate);


test('should render MainPage correctly', () => {
  const wrapper = shallow(<MainPage personData={JSON.stringify(persons)} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});