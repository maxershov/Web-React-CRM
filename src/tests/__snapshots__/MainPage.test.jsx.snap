// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MainPage correctly 1`] = `
<Fragment>
  <div
    className="mainPage"
  >
    <h1
      className="askPhoneTurn font_white_shadow"
    >
      Rotate screen to landscape mode
      <br />
      ⤵
    </h1>
    <Calendar
      className="calendar calendarMain"
      maxDetail="month"
      minDetail="century"
      onChange={[Function]}
      returnValue="start"
      showNavigation={true}
      showNeighboringMonth={true}
      value={1970-01-01T00:00:00.000Z}
      view="month"
    />
    <div
      className="notesMain"
    >
      <AreaNotes
        dayObject={
          Object {
            "date": "31-12-1969",
            "history": Array [],
            "notes": "",
          }
        }
        notesValue=""
        type="DAY_DATA"
      />
    </div>
    <div
      className="newProfileField"
    >
      <FormData
        baseValue=""
        formLabel="Новый профиль:"
        route={
          Object {
            "push": [MockFunction],
          }
        }
        type="NEW_PERSON"
      />
    </div>
    <div
      className="newCodeField"
    >
      <CodeScanner
        date="31-12-1969"
        dayObject={
          Object {
            "date": "31-12-1969",
            "history": Array [],
            "notes": "",
          }
        }
      />
    </div>
  </div>
  <div
    className="tableMain"
  >
    <t
      AggregatedComponent={[Function]}
      ExpanderComponent={[Function]}
      FilterComponent={[Function]}
      LoadingComponent={[Function]}
      NoDataComponent={[Function]}
      PadRowComponent={[Function]}
      PaginationComponent={[Function]}
      PivotValueComponent={[Function]}
      ResizerComponent={[Function]}
      TableComponent={[Function]}
      TbodyComponent={[Function]}
      TdComponent={[Function]}
      TfootComponent={[Function]}
      ThComponent={[Function]}
      TheadComponent={[Function]}
      TrComponent={[Function]}
      TrGroupComponent={[Function]}
      aggregatedKey="_aggregated"
      className="table -striped -highlight"
      collapseOnDataChange={true}
      collapseOnPageChange={true}
      collapseOnSortingChange={true}
      column={
        Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "Placeholder": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "",
          "headerStyle": Object {},
          "minResizeWidth": 11,
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        }
      }
      columns={
        Array [
          Object {
            "Cell": [Function],
            "Header": "Фото",
            "accessor": "code",
            "headerClassName": "tableHeader",
            "width": 205,
          },
          Object {
            "Cell": [Function],
            "Header": "Имя",
            "accessor": "code",
            "headerClassName": "tableHeader",
            "style": Object {
              "whiteSpace": "unset",
            },
            "width": 614,
          },
          Object {
            "Header": "Время",
            "accessor": "time",
            "headerClassName": "tableHeader",
            "width": 205,
          },
        ]
      }
      data={Array []}
      defaultExpanded={Object {}}
      defaultFilterMethod={[Function]}
      defaultFiltered={Array []}
      defaultPage={0}
      defaultPageSize={5}
      defaultResized={Array []}
      defaultSortDesc={false}
      defaultSortMethod={[Function]}
      defaultSorted={
        Array [
          Object {
            "desc": true,
            "id": "time",
          },
        ]
      }
      expanderDefaults={
        Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        }
      }
      filterable={false}
      freezeWhenExpanded={false}
      getLoadingProps={[Function]}
      getNoDataProps={[Function]}
      getPaginationProps={[Function]}
      getProps={[Function]}
      getResizerProps={[Function]}
      getTableProps={[Function]}
      getTbodyProps={[Function]}
      getTdProps={[Function]}
      getTfootProps={[Function]}
      getTfootTdProps={[Function]}
      getTfootTrProps={[Function]}
      getTheadFilterProps={[Function]}
      getTheadFilterThProps={[Function]}
      getTheadFilterTrProps={[Function]}
      getTheadGroupProps={[Function]}
      getTheadGroupThProps={[Function]}
      getTheadGroupTrProps={[Function]}
      getTheadProps={[Function]}
      getTheadThProps={[Function]}
      getTheadTrProps={[Function]}
      getTrGroupProps={[Function]}
      getTrProps={[Function]}
      groupedByPivotKey="_groupedByPivot"
      indexKey="_index"
      loading={false}
      loadingText="Загрузка"
      multiSort={true}
      nestingLevelKey="_nestingLevel"
      nextText="Вперед"
      noDataText="Нет данных"
      ofText="из"
      onFetchData={[Function]}
      originalKey="_original"
      pageJumpText="jump to page"
      pageSizeOptions={
        Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ]
      }
      pageText="Страница"
      pivotDefaults={Object {}}
      pivotIDKey="_pivotID"
      pivotValKey="_pivotVal"
      previousText="Назад"
      resizable={true}
      resolveData={[Function]}
      rowsSelectorText="rows per page"
      rowsText="профилей"
      showPageJump={true}
      showPageSizeOptions={true}
      showPagination={true}
      showPaginationBottom={true}
      showPaginationTop={false}
      sortable={true}
      style={Object {}}
      subRowsKey="_subRows"
    />
  </div>
</Fragment>
`;
